### PR TITLE
[NAVAJO-10] substitution component

### PR DESCRIPTION
**Pull request by amenders**
Closes: #10

### CODE REVIEW

File: src/renderer/components/Substitution.vue [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
-    name: 'substitution'
+    name: 'substitution',
+    components: {
+      cipherInputsSub
+    }
   }
 </script>
+<style lang="scss">
+  .view {
+    display: flex;
```
Comment (Jbcampbe) [inline src/renderer/components/Substitution.vue]: There's probably and standard way to share these kinds of styles between components in Vue. Not sure if external SCSS files is the way to go or what... Not necessarily asking for it to be changed (cuz I've done the same thing)... mostly making a note : )

File: src/renderer/components/cipher-inputs.vue [hunk 2]
```diff
@@ -50,7 +50,7 @@
 
   .cipher-input {
     resize: none;
-    height: 300px;
+    height: 150px;
```
Comment (Jbcampbe) [inline src/renderer/components/cipher-inputs.vue]: What do you think of making this `30 vh`? Testing it out on my computer it seems to work well at multiple different view sizes. With the fixed height of 150px it doesn't look so great on big screens.

screen shot 2018-04-12 at 6 04 59 pm

**Comment by Jbcampbe** [comment]
Looks like the plaintext div needs some width constraints

screen shot 2018-04-12 at 6 31 08 pm

**Comment by Jbcampbe** [comment]
I think we want to enforce no duplicate keys in the letter mappings

screen shot 2018-04-12 at 6 34 01 pm

**Comment by Jbcampbe** [comment]
Core functionality-wise though it looks good : )

File: src/renderer/components/Substitution.vue (hunk 1 above)
Comment (amenders) [reply src/renderer/components/Substitution.vue]: Yeah, I'm sure there MUST be a way to do that...

File: src/renderer/components/cipher-inputs-sub.vue [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      },
+
+      updateMap (letter, e) {
+        this.map[letter] = e.target.value
+
+        var values = Object.values(this.map)
+        var keys = Object.keys(this.map)
+        for (var i = 0; i < Object.values(this.map).length; i++) {
+          if (values.indexOf(values[i]) !== values.lastIndexOf(values[i]) && values[i] !== '') {
+            console.log('true')
```
Comment (Jbcampbe) [inline src/renderer/components/cipher-inputs-sub.vue]: we can probably clean these up now

**Comment by Jbcampbe** [comment]
Sorry. Still seeing this behavior : /

screen shot 2018-04-20 at 11 13 17 am

**Review by Jbcampbe — APPROVED** [review]

**Pull request merged**